### PR TITLE
More Standard Library Improvements

### LINKED ISSUE #1279: `File.open` method
### Summary
While our `File` library provides users with good levels of control, it is somewhat lacking in terms of ease of use. We want to add a `File.open -> Text -> Text` that opens and reads the file in one shot.

### Value
Files can be opened and read in one shot.

### Specification

- [x] Add a `File.open` method that opens and reads the file in one go.

### Acceptance Criteria & Test Cases

- [x] Files can be opened and read in one function call.

### PR DESCRIPTION
**Pull request by iamrecursion**
### Pull Request Description
Fixes a bunch of stuff and adds new features.

Closes #1279.

### Important Notes
N/A

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: distribution/std-lib/Base/src/Data/Any/Extensions.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Any/Extensions.enso]: Remove this method – too much methods on `Any` and a clumsy name.

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso]: Useless example – it's exactly the same as just pressing enter in the searcher

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso]: Unnecessary recursion, use `vec.fold` instead.

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso]: write it as `this.is_empty.not` – less work when we migrate to non-UFCS

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso]:
```suggestion
    get : Any -> Any | Nothing
```

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso:150]: Write this from scratch – you're allocating a copy of this map just to drop it.

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso]: `Fold over the values` means nothing to a non-functional non-dev. Reword with something like "combine all values..."

File: distribution/std-lib/Base/src/Data/Map.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Map.enso]: Same comment as `fold`

File: distribution/std-lib/Base/src/Data/Range.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Range.enso]: stray comment

File: distribution/std-lib/Base/src/Data/Text/Extensions.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Text/Extensions.enso:126]: `non_empty` seems nicer to me. Your call tho :)

File: distribution/std-lib/Base/src/Data/Time/Locale.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Time/Locale.enso]: What's the key to this assortment of countries? Seems pretty strange to me. Misses some large countries (India, Russia, Brazil) while exposing, e.g. Taiwan. I'd go by countries-with-highest-population (or countries with highest population + countries with highest GDP, or some other metric like that)

File: distribution/std-lib/Base/src/Data/Time/Locale.enso
Comment (kustosz) [inline distribution/std-lib/Base/src/Data/Time/Locale.enso]: forgot to change `lang` in these copy-pastes

**Review by kustosz — APPROVED** [review]
Very cool additions! There are some problems that must be addressed before merge, but I'm approving because they don't require a re-review.

**Review by 4e6 — APPROVED** [review]
LGTM

**Review by kustosz — APPROVED** [review]
Looks good, one stray comment left :)

**Pull request merged**
